packages/config/src/discovery/ProjectDiscovery.ts

```
@@ -49,60 +50,33 @@ const paths = getDiscoveryPaths()
+  private readonly reachableEntries: EntryParameters[]
   private eoaIDMap: Record<string, string> = {}
   private permissionRegistry: PermissionRegistry
 
   constructor(
     public readonly projectName: string,
     public readonly configReader = new ConfigReader(paths.discovery),
   ) {
-    this.discoveries = []
-    let dependentDiscoveries = {}
     try {
-      const projectDiscovery = configReader.readDiscovery(projectName)
-      this.discoveries.push(projectDiscovery)
-      dependentDiscoveries = projectDiscovery.dependentDiscoveries ?? {}
-    } catch {}
-
-    const alreadyAdded = new Set<string>(this.discoveries.map((x) => x.name))
-    for (const discovery of this.discoveries) {
-      const referencedProjects = discovery.entries
-        .map((e) => e.targetProject)
-        .filter(notUndefined)
-
-      // TODO: necessary until we remove .sharedModules completely
-      const allReferencedProjects = [
-        ...referencedProjects,
-        ...(discovery.sharedModules ?? []),
-      ]
-
-      for (const sharedModule of allReferencedProjects) {
-        const key = sharedModule
-        if (alreadyAdded.has(key)) continue
-
-        try {
-          this.discoveries.push(configReader.readDiscovery(sharedModule))
-        } catch {}
-        alreadyAdded.add(key)
-      }
+      this.discoveries = configReader.readDiscoveryWithReferences(projectName)
+      const entrypoints = [...this.discoveries[0].entries].map((e) => e.address) // always the base discovery
+
+      // Removing Reference entries because otherwise we get duplicates
+      // and incomplete data.
+      // TODO: refactor this whole logic around depenent projects and
+      // references to entrypoints to make it cleaner
+      this.discoveries.forEach((d) => removeReferences(d))
+
+      this.reachableEntries = getReachableEntries(
+        this.discoveries.flatMap((discovery) => discovery.entries),
+        entrypoints,
+      )
+    } catch {
```

We shouldn't silently fail. Why is this necessary?

---

Legacy behavior - wicked tho, since tests fail all over the place. High time to fix it, I guess. Related to non-existing discoveries when making new instance of PD